### ISSUE TITLE
`#[deny(missing_docs)]` ignores exported `macro_rules!`

### ISSUE THREAD

**Issue by kaikalii** [rust-lang/rust#56334]
If my crate exports a `macro_rules!` macro, the `deny(missing_docs)` attribute has no effect even though it probably should.

The following code should, at the very least, provide a warning
```rust
#[deny(missing_docs)]
#[macro_export]
macro_rules! foo {
    () => {}
}
```

Preventing this code from compiling would likely break some backwards compatability, but I don't see why it shouldn't at least give a warning.

**Comment by Centril** [comment]
cc @petrochenkov  @rust-lang/lang I cannot decide whether this is expected behavior or not, but it makes sense that `missing_docs` should fire on exported macros.

**Comment by cramertj** [comment]
> Preventing this code from compiling would likely break some backwards compatability

We don't guarantee backwards-compatibility for `#![deny(warnings)]`, otherwise we would never be able to introduce a new warning. I agree that missing `deny(missing_docs)` should error on crate-exported macros with no documentation.

**Comment by scottmcm** [comment]
Lang team discussion agreed with the two comments above; the lint should warn for this.

**Comment by nnethercote** [comment]
This broke compilation of existing code in the rustc-perf benchmark suite, see https://github.com/rust-lang-nursery/rustc-perf/pull/325

I see the comment "We don't guarantee backwards-compatibility for `#![deny(warnings)]`" above, but I thought it might be useful to note some practical consequences.
